### PR TITLE
Help greenkeeper to manage lockfiles

### PR DESCRIPTION
**Pull request by harrysarson**
Fixes issues we are having with greenkeeper, `npm ci`, and lock files.

Solution based on https://github.com/greenkeeperio/greenkeeper-lockfile#npm.

Should prevent CI failures in greenkeeper's PR #1183.
Greenkeeper issue: https://github.com/greenkeeperio/greenkeeper-lockfile/issues/156.

@josdejong How tight should we be on the npm and greenkeeper-lock file versions?

### CODE REVIEW

**Comment by josdejong** [comment]
> @josdejong How tight should we be on the npm and greenkeeper-lock file versions?

I'm not sure, what are our options in this regard? To me the package-lock.json file is something that's just there to make sure that when you or I install dependencies at some point after checking out the latest version of a branch or some specific tag or commit, that we get installed the exact same versions of all dependencies. We can update the package-lock.json file as often as we want/need.

Your PR looks good to me, can you merge this PR and the related greenkeeper PR's?

File: .travis.yml [hunk 1]
```diff
@@ -9,8 +9,18 @@ stages:
   - test
   - browser-test
 
+before_install:
+  - npm install -g npm@6
+  - npm install -g greenkeeper-lockfile@1
```
Comment (harrysarson) [inline .travis.yml]: @josdejong In these two lines I install npm and greenkeeper-lock file. Currently, we install the latest minor version. Should we be more specific?

**Comment by harrysarson** [comment]
Once we clarify the versions for npm install in https://github.com/josdejong/mathjs/pull/1189#pullrequestreview-141217694 I will merge :+1:

File: .travis.yml (hunk 1 above)
Comment (josdejong) [reply .travis.yml]: I like it like this. 

If we want to be more specific some day we can always change it. But then we need some greenkeeper solution to keep the `.travis.yml` file up to date too ;)

Comment (josdejong) [reply .travis.yml]: ping @harrysarson

Comment (harrysarson) [reply .travis.yml]: Greenkeeper-ception :laugh:

**Comment by josdejong** [comment]
:+1:

**Pull request merged**